src/Stratis.Bitcoin/Utilities/RetryOptions.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Linq;
+
+namespace Stratis.Bitcoin.Utilities
+{
+    public class RetryOptions
+    {
+        public RetryOptions()
+        {
+            this.RetryCount = 1;
+            this.Delay = TimeSpan.FromMilliseconds(100);
+            this.ExceptionTypes = new[] { typeof(Exception) };
+        }
+
+        public RetryOptions(short retryCount, TimeSpan delay, params Type[] exceptionTypeTypes)
+        {
+            foreach (Type exceptionType in exceptionTypeTypes)
+            {
+                if (!typeof(Exception).IsAssignableFrom(exceptionType)) 
+                {
+                    throw new ArgumentException($"All exception types must be valid exceptions. {exceptionType} is not an Exception.");
+                }
+            }
+
+            if (retryCount < 1)
+                throw new ArgumentOutOfRangeException(nameof(retryCount), "Retry count cannot be less or equal to 0.");
+
+            if (exceptionTypeTypes.Length == 0)
+            {
+                this.ExceptionTypes = new[] { typeof(Exception) };
```

Comment:
this gets overwritten later, but in any case we can not add a default of `Exception`, keep the list empty if  `exceptionTypeTypes.Length == 0`

Reply:
Good pickup, removed default